v2/bundleendpoint/fetch.go

```diff
@@ -0,0 +1,24 @@
+package bundleendpoint
+
+import (
+	"context"
+
+	"github.com/spiffe/go-spiffe/v2/spiffebundle"
+	"github.com/spiffe/go-spiffe/v2/spiffetls"
+	"github.com/spiffe/go-spiffe/v2/spiffex509"
+)
+
+// FetchOption is an option used when dialing the bundle endpoint.
+type FetchOption interface{}
```

Comment:
lI wasn't sure when i wrote this part if we'd need options exclusive to watching or serving so I prefixed with `Fetch`.... I'd be ok renaming to `Option` if we like the rest of the package and don't see it growing (i.e. eventually adding options to `Handler()`, like `Handler(source Source, options ...HandlerOption)`). And yes, this can move to option.go to be consistent with other packages.